### PR TITLE
Initial TypeScript port

### PR DESCRIPTION
**Pull request by johnnyreilly**
This is not ready for merging yet but I thought you'd like to see the first steps on #45... (Also I'd like to see if the tests still pass - still can't run on Windows!)

### CODE REVIEW

**Comment by johnnyreilly** [comment]
The tests pass on first attempt - hah!  Check me out 😉 

I did not expect it to work first time.  Quite pleased it did.  I'd planned to remove the `lib` directory and leave only the `src` directory committed. (The `lib` directory is generated from the `src` directory now.)  What do you think?

**Comment by piotr-oles** [comment]
You can remove it but you have to add `prepublish` script in `package.json` to build lib before release. Also you will have to update tests to compile before run and add `lib` to `.gitignore` :)

**Comment by johnnyreilly** [comment]
Cool - doing it

**Comment by johnnyreilly** [comment]
Done

**Comment by johnnyreilly** [comment]
OK - I've made it `noImplicitAny: true` compliant (can't stand an `any` 😄).  Feel free to merge!

File: src/NormalizedMessage.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      return 1;
+    }
+
+    return (
+      NormalizedMessage.compareTypes(messageA.getType(), messageB.getType()) ||
+      NormalizedMessage.compareOptionalStrings(messageA.getFile(), messageB.getFile()) ||
+      NormalizedMessage.compareSeverities(messageA.getSeverity(), messageB.getSeverity()) ||
+      NormalizedMessage.compareNumbers(messageA.getLine(), messageB.getLine()) ||
+      NormalizedMessage.compareNumbers(messageA.getCharacter(), messageB.getCharacter()) ||
+      // code can be string (lint failure) or number (typescript error) - should the following line cater for this in some way?
```
Comment (johnnyreilly) [inline src/NormalizedMessage.ts:92]: I think this might be a bug?  I've left it as is just in case.

Comment (johnnyreilly) [reply src/NormalizedMessage.ts:92]: Line 93 I mean - 92 is the comment I put in relating to this...

**Comment by piotr-oles** [comment]
Ok, I've checked it and looks good :) I will not release it now because it doesn't add a new functionality. I plan to add errors limit option to avoid out of memory on big projects refactoring, so it will be released then :)

**Comment by johnnyreilly** [comment]
Great!

**Pull request merged**